### ISSUE TITLE
How to use it with Node ?

### ISSUE THREAD

**Issue by localhero44** [keyboardio/teensy-loader.js#3]
Hi,
First thank you for this project as it is a great way to simplify the hex upload on teensy.
I've tried to make it work with NodeJS but it fails with the module syntax.
Could you provide a full example please ?
I am trying to upload to a Teensy 3.2, with slight modifications of the code.
Regards
David

**Comment by algernon** [comment]
The example was set up with Babel in mind. For vanilla NodeJS, something like this should work:

```js
TeensyLoader = require('teensy-loader').default;

TeensyLoader.upload(0x16C0, 0x0478, "./firmware.hex", function (addr, size) {
  process.stdout.write(".");
});
```

Hope this helps!

PS: If you make it work with Teensy 3.2, it'd be lovely if you could send a pull request with your changes :)

**Comment by localhero44** [comment]
Thanks. No luck with this 'require' syntax.
Got it to launch with the "esm" module for Node.
Now it launches but can't find the device at this line : 
`usb.findByIds(vid, pid) (with 0x16C0, 0x0478).`
When I add a log when calling `usb.getDeviceList` I can see a device with this vid, but not the same pid : 
`{"busNumber":2,"deviceAddress":6,"deviceDescriptor":{"bLength":18,"bDescriptorType":1,"bcdUSB":272,"bDeviceClass":2,"bDeviceSubClass":0,"bDeviceProtocol":0,"bMaxPacketSize0":64,"idVendor":5824,"idProduct":1155,"bcdDevice":629,"iManufacturer":1,"iProduct":2,"iSerialNumber":3,"bNumConfigurations":1},"portNumbers":[6]},`

Strange.
I've tried replace 0x478 with 0x483 (1155 just above), but then I get this error :
`__open findByIds: 5824/ pid: 1155
(node:7076) UnhandledPromiseRejectionWarning: Error: LIBUSB_ERROR_NOT_SUPPORTED`

**Comment by algernon** [comment]
Not sure I can help then, I'm afraid, I have no Teensy 3.2s to test with :(

**Comment by localhero44** [comment]
I've tried something. One way to investigate.
The device 0x0478 isn't found until I push the button on the Teensy. Then it gives something like that : 
```
__open device:undefined
__open device:undefined
__open device:undefined
__open device:undefined
__open device:undefined
__open device:undefined
__open device:{"busNumber":2,"deviceAddress":49,"deviceDescriptor":{"bLength":18,"bDescriptorType":1,"bcdUSB":272,"bDeviceClass":0,"bDeviceSubClass":0,"bDeviceProtocol":0,"bMaxPacketSize0":64,"idVendor":5824,"idProduct":1144,"bcdDevice":259,"iManufacturer":0,"iProduct":0,"iSerialNumber":1,"bNumConfigurations":1},"portNumbers":[14]}
```
Then it seems to send the hex file, but not properly I guess as the Teensy never reboot and never blinks :(
With the teensy_loader_cli with the same HEX file it works.
I need to monitor what is sent on the COM port/

**Comment by algernon** [comment]
Oh, right. The Teensy 3.2 uses different block sizes as far as I remember. I might be able to do something about that, but no ETA at the moment, as my hands are full with other tasks.

**Comment by localhero44** [comment]
That's right, but I've changed that as soon as I had downloaded your code : 
```
      code_size: 262144,
      block_size: 1024
```

**Comment by localhero44** [comment]
Hi, I have tried again after a long time away of this project.
Still no luck, I had make changes in the code to add a 64 bytes offset from what I see in the C code from PJRC : 
```
 else if (block_size == 512 || block_size == 1024) {
			buf[0] = addr & 255;
			buf[1] = (addr >> 8) & 255;
			buf[2] = (addr >> 16) & 255;
			memset(buf + 3, 0, 61);
			ihex_get_data(addr, block_size, buf + 64);
			write_size = block_size + 64;
		}
```

That sends 1088 bytes blocks (1024 + 64) but still nothing happens on the Teensy.
How could I check what the CLI loader in C sends with the same HEX file ?
Cheers
David
